cmd/mimir/Dockerfile.distroless

```
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: AGPL-3.0-only
+
+FROM       gcr.io/distroless/base-nossl-debian12
```

Can we switch to using static-debian12 by default, and only use `gcr.io/distroless/base-nossl-debian12` when for builds with race-detector enabled?

```suggestion
ARG          BASEIMG=gcr.io/distroless/static-debian12
FROM       $BASEIMG
```